avocado/core/test.py

```diff
@@ -973,6 +973,18 @@ def test(self):
                 os.chdir(pre_cwd)
 
 
+class PythonUnittest(ExternalRunnerTest):
+    """
+    Python unittest test
+    """
+    def __init__(self, name, params=None, base_logdir=None, job=None,
+                 test_dir=None):
+        runner = "%s -m unittest -q -c" % sys.executable
```

Comment:
TBH, I was hoping for a way to not base `PythonUnittest` on `ExternalRunnerTest`.  I tried it briefly before, and failed because of the deep changes that seemed to be required.  So, yes, this is definitely better than what we have right now.  I was doing some profiling, and it looks like starting a Python process and the unittest takes on average 1 second per test, on a not too simplistic test:

```
$ /usr/bin/time python -m unittest selftests.unit.test_replay.Replay.test_replay_map_interrupted_json; /usr/bin/time python -m unittest selftests.unit.test_replay.Replay.test_replay_map_after_crash; /usr/bin/time python -m unittest selftests.unit.test_replay.Replay.test_tap_parsing; 
.
----------------------------------------------------------------------
Ran 1 test in 0.004s

OK
1.04user 0.11system 0:01.16elapsed 99%CPU (0avgtext+0avgdata 38912maxresident)k
0inputs+0outputs (0major+6585minor)pagefaults 0swaps
.
----------------------------------------------------------------------
Ran 1 test in 0.005s

OK
1.04user 0.12system 0:01.17elapsed 99%CPU (0avgtext+0avgdata 39204maxresident)k
0inputs+0outputs (0major+6589minor)pagefaults 0swaps
.
----------------------------------------------------------------------
Ran 1 test in 0.005s

OK
1.03user 0.11system 0:01.16elapsed 99%CPU (0avgtext+0avgdata 39088maxresident)k
0inputs+0outputs (0major+6592minor)pagefaults 0swaps
```

Which is not that far from the results when running the same tests with this Avocado "wrapper":

```
$ /usr/bin/time avocado run selftests/unit/test_replay.py 
JOB ID     : 1b0c56b1c8a0edbfae161d4350a73ccb397e0cd1
JOB LOG    : /home/cleber/avocado/job-results/job-2017-09-12T13.25-1b0c56b/job.log
 (1/3) test_replay.Replay.test_replay_map_interrupted_json: PASS (1.21 s)
 (2/3) test_replay.Replay.test_replay_map_after_crash: PASS (1.21 s)
 (3/3) test_replay.Replay.test_tap_parsing: PASS (1.21 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 4.00 s
JOB HTML   : /home/cleber/avocado/job-results/job-2017-09-12T13.25-1b0c56b/results.html
5.20user 0.78system 0:06.13elapsed 97%CPU (0avgtext+0avgdata 76108maxresident)k
0inputs+1296outputs (0major+72031minor)pagefaults 0swaps
```

We all know there's a lot of room for optimization, but I'm more worried about the loader shortcomings I mentioned in another comment.

Reply:
Actually I keep saying the same. It's not because of ExternalRunnerTest, but because any test (even `--dry-run`) takes unbelievably long.

Just for record my downclocked W541 to 800MHz:
passtest.py: 0.04s
passtest.sh: 0.04s
b.Foo (simple PASS unittest): 0.10s

Anyway yes, we can eventually create unittest runner and use it instead of running another process.